elephant/spike_train_processing.py

```diff
@@ -0,0 +1,317 @@
+        # defined by the boundaries
+        # note that every second entry in boundaries is an upper boundary
+        spike_to_epoch_idx = np.searchsorted(right_edges,
+                                             st.times.rescale(
+                                                 right_edges.units))
+        complexity_per_spike = complexity[spike_to_epoch_idx]
+
+        st.array_annotate(complexity=complexity_per_spike)
+
+        if deletion_threshold is not None:
+            mask = complexity_per_spike < deletion_threshold
+            if invert_delete:
+                mask = np.invert(mask)
+            old_st = st
+            new_st = old_st[mask]
+            spiketrains[idx] = new_st
+            unit = old_st.unit
+            segment = old_st.segment
+            if unit is not None:
+                unit.spiketrains[_get_index(unit.spiketrains,
+                                            old_st)] = new_st
+            if segment is not None:
+                segment.spiketrains[_get_index(segment.spiketrains,
+                                               old_st)] = new_st
+            del old_st
+
+    return complexity_epoch
+
+
+def precise_complexity_intervals(spiketrains, sampling_rate, spread=0):
+    """
+    Calculate the complexity (i.e. number of synchronous spikes found)
+    at `sampling_rate` precision in a list of spiketrains.
+
+    Complexity is calculated by counting the number of spikes (i.e. non-empty
+    bins) that occur separated by `spread - 1` or less empty bins, within and
+    across spike trains in the `spiketrains` list.
+
+    Parameters
+    ----------
+    spiketrains : list of neo.SpikeTrains
+        a list of neo.SpikeTrains objects. These spike trains should have been
+        recorded simultaneously.
+    sampling_rate : pq.Quantity
+        Sampling rate of the spike trains.
+    spread : int, optional
+        Number of bins in which to check for synchronous spikes.
+        Spikes that occur separated by `spread - 1` or less empty bins are
+        considered synchronous.
+          * `spread = 0` corresponds to a bincount accross spike trains.
+          * `spread = 1` corresponds to counting consecutive spikes.
+          * `spread = 2` corresponds to counting consecutive spikes and
+            spikes separated by exactly 1 empty bin.
+          * `spread = n` corresponds to counting spikes separated by exactly
+            or less than `n - 1` empty bins.
+        Default: 0
+
+    Returns
+    -------
+    complexity_intervals : neo.Epoch
+        An epoch object containing complexity values, left edges and durations
+        of all intervals with at least one spike.
+
+    Raises
+    ------
+    ValueError
+        When `t_stop` is smaller than `t_start`.
+
+    Examples
+    --------
+    Here the behavior of
+    `elephant.spike_train_processing.precise_complexity_intervals` is shown, by
+    applying the function to some sample spiketrains.
```

Comment:
The description of an example is optional and not required if there is nothing more to say than the code speaks by itself.